docker/ubuntu_22.04-dev.dockerfile

```diff
@@ -65,30 +74,52 @@ RUN conda uninstall -y conda-libmamba-solver
 RUN conda config --set solver classic
 RUN conda update -y --all && \
     mamba install -y \
-                cmake \
+                make \
+                gxx_linux-64 \
+                gcc_linux-64 \
                 git \
+                cmake \
+                gfortran \ 
                 libblas \
                 liblapack \
+                eigen \
                 hdf5 \
-                setuptools \
+                numpy \
+                scipy \
+                "cython<3" \
+                nose \
                 pytest \
                 pytables \
+                matplotlib \
                 jinja2 \
-                "cython<3" \
+                setuptools \
+                future \
+                progress \
                 && \
     mamba install -y --force-reinstall libsqlite && \
     conda clean -y --all
 RUN mkdir -p `python -m site --user-site`
 
+ENV CC /opt/conda/bin/x86_64-conda_cos6-linux-gnu-gcc
+ENV CXX /opt/conda/bin/x86_64-conda_cos6-linux-gnu-g++
+ENV CPP /opt/conda/bin/x86_64-conda_cos6-linux-gnu-cpp
+
+# put conda on the path
+ENV LD_LIBRARY_PATH /opt/conda/lib:$LD_LIBRARY_PATH
+# set HDF5 path for base_python stage
+ENV HDF5_INSTALL_PATH=/opt/conda
+
+
 FROM ${pkg_mgr}_deps AS base_python
 # make starting directory
 RUN mkdir -p $HOME/opt
 RUN echo "export PATH=$HOME/.local/bin:\$PATH" >> ~/.bashrc
 
 # build HDF5
 ARG build_hdf5="NO"
-ENV HDF5_INSTALL_PATH=$HOME/opt/hdf5/$build_hdf5
-RUN if [ "$build_hdf5" != "NO" ]; then \
+ARG pkg_mgr=apt
+# don't build hdf5 if we use conda (we already installed it)
```

Comment:
We install HDF5 in the `apt` path already, too.  I think we do want to build in both cases.  However, when we choose not to build (`build_hdf5 = NO`) then it fails, which is a different problem.  It looks like it's using HDF 1.14 when we install via conda - not sure we've tested PyNE with that before.

Reply:
HDF5 and MOAB are dependencies of PyNE to enable all functionalities. When building a conda package, it is advisable to rely on the conda versions of HDF5 and MOAB. This ensures that when a user installs PyNE through conda, HDF5 and MOAB will be automatically installed for them.